[Spring Cloud Datastore] Can't read Blob type into byte[] using Repository and Entity

**Describe the bug**
Please provide details of the problem, including the version of Spring Cloud that you
are using. 

```
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>2.5.5</version>
        <relativePath/> <!-- lookup parent from repository -->
    </parent>
    <groupId>xxxxxxxxxxx</groupId>
    <artifactId>xxxxxxxx</artifactId>
    <version>0.0.1-SNAPSHOT</version>
    <name>backend</name>
    <description>xxxxxxxxxxxxxxxxxx</description>

    <properties>
        <java.version>1.8</java.version>
    </properties>

    <dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>com.google.cloud</groupId>
                <artifactId>spring-cloud-gcp-dependencies</artifactId>
                <version>2.0.4</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>

    <dependencies>

        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-data-rest</artifactId>
        </dependency>

        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-devtools</artifactId>
            <scope>runtime</scope>
            <optional>true</optional>
        </dependency>

        <dependency>
            <groupId>junit</groupId>
            <artifactId>junit</artifactId>
            <version>4.12</version>
        </dependency>

        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-test</artifactId>
            <scope>test</scope>
            <exclusions>
                <exclusion>
                    <groupId>org.junit.vintage</groupId>
                    <artifactId>junit-vintage-engine</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>com.auth0</groupId>
            <artifactId>java-jwt</artifactId>
            <version>3.10.3</version>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-security</artifactId>
            <version>2.3.3.RELEASE</version>
        </dependency>
        <dependency>
            <groupId>com.amazonaws</groupId>
            <artifactId>aws-java-sdk</artifactId>
            <version>1.11.918</version>
        </dependency>
        <dependency>
            <groupId>com.google.cloud</groupId>
            <artifactId>spring-cloud-gcp-starter-data-datastore</artifactId>
        </dependency>
    </dependencies>

    <build>
        <plugins>
            <plugin>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-maven-plugin</artifactId>
            </plugin>
        </plugins>
    </build>

</project>
```
I have an entity of Admin:

```
@Entity
public class Admin implements Role {
    @Id
    Key key;
    private UUID id;
    private String username;
    private String password;
    private LocalDateTime tokenExpired;
    private byte[] tokenSecret;
}

```
And the following AdminRepository:

```
public interface AdminRepository extends DatastoreRepository<Admin, Key> {
    Optional<Admin> findOneByUsername(String username);
}
```

At first, I tried to save the entity to Datastore using `adminRepository.save(admin)`, it was fine and a record was generated in the database. The `byte[]` type was automatically converted to `Blob`. However, when I called `adminRepository.findOneByUsername(username)` to get the record I previously inserted, it always throughs an exception.

```
Caused by: com.google.cloud.spring.data.datastore.core.mapping.DatastoreDataException: Unable to read property tokenSecret; nested exception is org.springframework.core.convert.ConverterNotFoundException: No converter found capable of converting from type [com.google.cloud.datastore.Blob] to type [java.lang.Byte]
	at com.google.cloud.spring.data.datastore.core.convert.EntityPropertyValueProvider.getPropertyValue(EntityPropertyValueProvider.java:75)
	at com.google.cloud.spring.data.datastore.core.convert.EntityPropertyValueProvider.getPropertyValue(EntityPropertyValueProvider.java:59)
	at com.google.cloud.spring.data.datastore.core.convert.DefaultDatastoreEntityConverter.lambda$read$0(DefaultDatastoreEntityConverter.java:170)
	at com.google.cloud.spring.data.datastore.core.mapping.DatastorePersistentEntityImpl.lambda$doWithColumnBackedProperties$0(DatastorePersistentEntityImpl.java:206)
	at org.springframework.data.mapping.model.BasicPersistentEntity.doWithProperties(BasicPersistentEntity.java:360)
	at com.google.cloud.spring.data.datastore.core.mapping.DatastorePersistentEntityImpl.doWithColumnBackedProperties(DatastorePersistentEntityImpl.java:203)
	at com.google.cloud.spring.data.datastore.core.convert.DefaultDatastoreEntityConverter.read(DefaultDatastoreEntityConverter.java:166)
	... 67 more
Caused by: org.springframework.core.convert.ConverterNotFoundException: No converter found capable of converting from type [com.google.cloud.datastore.Blob] to type [java.lang.Byte]
	at org.springframework.core.convert.support.GenericConversionService.handleConverterNotFound(GenericConversionService.java:322)
	at org.springframework.core.convert.support.GenericConversionService.convert(GenericConversionService.java:195)
	at org.springframework.core.convert.support.GenericConversionService.convert(GenericConversionService.java:175)
	at com.google.cloud.spring.data.datastore.core.convert.TwoStepsConversions.convertOnReadSingle(TwoStepsConversions.java:220)
	at com.google.cloud.spring.data.datastore.core.convert.TwoStepsConversions.convertOnRead(TwoStepsConversions.java:178)
	at com.google.cloud.spring.data.datastore.core.convert.TwoStepsConversions.convertOnRead(TwoStepsConversions.java:130)
	at com.google.cloud.spring.data.datastore.core.convert.EntityPropertyValueProvider.getPropertyValue(EntityPropertyValueProvider.java:71)
	... 73 more
```
 It seems that the `Blob` object can't be converted back to `byte[]` here. However, the document of ` 14.2.5. Supported Types` said that the `byte[]` is supported in Datastore. It should be successfully converted. Also, the error message said that `Blob` can't be converted to `Byte`. However, the type defined in my entity is `byte[]`, which is very confusing. I am very appreciated if someone can help me with this issue.

--- a/spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/convert/TwoStepsConversions.java
+++ b/spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/convert/TwoStepsConversions.java
@@ -120,7 +120,7 @@ public class TwoStepsConversions implements ReadWriteConversions {
 	public <T> T convertOnRead(Object val, EmbeddedType embeddedType, TypeInformation targetTypeInformation) {
 		TypeInformation componentTypeInformation;
 		Class collectionType = null;
-		if (targetTypeInformation.isCollectionLike()) {
+		if (ValueUtil.isCollectionLike(targetTypeInformation.getType())) {
 			componentTypeInformation = targetTypeInformation.getComponentType();
 			collectionType = targetTypeInformation.getType();
 		}
@@ -175,9 +175,6 @@ public class TwoStepsConversions implements ReadWriteConversions {
 			catch (ConversionException | DatastoreDataException ex) {
 				throw new DatastoreDataException("Unable process elements of a collection", ex);
 			}
-		} else if (val.getClass().equals(Blob.class) && targetCollectionType == byte[].class) {
-			// Special case where a single value (Blob) becomes a collection (byte[]).
-			return (T)((Blob) val).toByteArray();
 		}
 		// Convert single value.
 		return (T) readConverter.apply(val, targetComponentType);

--- a/spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/convert/TwoStepsConversions.java
+++ b/spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/convert/TwoStepsConversions.java
@@ -152,38 +152,35 @@ public class TwoStepsConversions implements ReadWriteConversions {
 			throw new DatastoreDataException(
 					"Unexpected property embedded type: " + embeddedType);
 		}
+
 		if (ValueUtil.isCollectionLike(val.getClass())
 				&& targetCollectionType != null && targetComponentType != null) {
 			// Convert collection.
-			return convertCollectionOnRead(val, targetCollectionType, targetComponentType, readConverter);
+			try {
+				List elements;
+				if (val.getClass().isArray()) {
+					elements = Collections.singletonList(val);
+				}
+				else {
+					elements = StreamSupport
+							.stream(((Iterable<?>) val).spliterator(), false)
+							.map(v -> {
+								Object o = (v instanceof Value) ? ((Value) v).get() : v;
+								return readConverter.apply(o, targetComponentType);
+							})
+							.collect(Collectors.toList());
+				}
+				return (T) convertCollection(elements, targetCollectionType);
+			}
+			catch (ConversionException | DatastoreDataException ex) {
+				throw new DatastoreDataException("Unable process elements of a collection", ex);
+			}
 		} else if (val.getClass().equals(Blob.class) && targetCollectionType == byte[].class) {
 			// Special case where a single value (Blob) becomes a collection (byte[]).
 			return (T)((Blob) val).toByteArray();
 		}
 		// Convert single value.
 		return (T) readConverter.apply(val, targetComponentType);
-	}
-
-	private <T> T convertCollectionOnRead(Object val, Class targetCollectionType, TypeInformation targetComponentType, BiFunction<Object, TypeInformation<?>, ?> readConverter) {
-		try {
-			List elements;
-			if (val.getClass().isArray()) {
-				elements = Collections.singletonList(val);
-			}
-			else {
-				elements = StreamSupport
-						.stream(((Iterable<?>) val).spliterator(), false)
-						.map(v -> {
-							Object o = (v instanceof Value) ? ((Value) v).get() : v;
-							return readConverter.apply(o, targetComponentType);
-						})
-						.collect(Collectors.toList());
-			}
-			return (T) convertCollection(elements, targetCollectionType);
-		}
-		catch (ConversionException | DatastoreDataException ex) {
-			throw new DatastoreDataException("Unable process elements of a collection", ex);
-		}
 	}
 
 	private <T, R> Map<T, R> convertOnReadSingleEmbeddedMap(Object value,

--- a/spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/convert/TwoStepsConversions.java
+++ b/spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/convert/TwoStepsConversions.java
@@ -155,27 +155,34 @@ public class TwoStepsConversions implements ReadWriteConversions {
 
 		if (ValueUtil.isCollectionLike(val.getClass())
 				&& targetCollectionType != null && targetComponentType != null) {
-			try {
-				List elements;
-				if (val.getClass().isArray()) {
-					elements = Collections.singletonList(val);
-				}
-				else {
-					elements = StreamSupport
-							.stream(((Iterable<?>) val).spliterator(), false)
-							.map(v -> {
-								Object o = (v instanceof Value) ? ((Value) v).get() : v;
-								return readConverter.apply(o, targetComponentType);
-							})
-							.collect(Collectors.toList());
-				}
-				return (T) convertCollection(elements, targetCollectionType);
-			}
-			catch (ConversionException | DatastoreDataException ex) {
-				throw new DatastoreDataException("Unable process elements of a collection", ex);
-			}
+			return convertCollectionOnRead(val, targetCollectionType, targetComponentType, readConverter);
+		} else if (val.getClass().equals(Blob.class) && targetCollectionType == byte[].class) {
+			return (T)((Blob) val).toByteArray();
 		}
 		return (T) readConverter.apply(val, targetComponentType);
+	}
+
+	private <T> T convertCollectionOnRead(Object val, Class targetCollectionType, TypeInformation targetComponentType, BiFunction<Object, TypeInformation<?>, ?> readConverter) {
+		try {
+			List elements;
+
+			if (val.getClass().isArray()) {
+				elements = Collections.singletonList(val);
+			}
+			else {
+				elements = StreamSupport
+						.stream(((Iterable<?>) val).spliterator(), false)
+						.map(v -> {
+							Object o = (v instanceof Value) ? ((Value) v).get() : v;
+							return readConverter.apply(o, targetComponentType);
+						})
+						.collect(Collectors.toList());
+			}
+			return (T) convertCollection(elements, targetCollectionType);
+		}
+		catch (ConversionException | DatastoreDataException ex) {
+			throw new DatastoreDataException("Unable process elements of a collection", ex);
+		}
 	}
 
 	private <T, R> Map<T, R> convertOnReadSingleEmbeddedMap(Object value,

--- a/spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/convert/TwoStepsConversions.java
+++ b/spring-cloud-gcp-data-datastore/src/main/java/com/google/cloud/spring/data/datastore/core/convert/TwoStepsConversions.java
@@ -152,20 +152,21 @@ public class TwoStepsConversions implements ReadWriteConversions {
 			throw new DatastoreDataException(
 					"Unexpected property embedded type: " + embeddedType);
 		}
-
 		if (ValueUtil.isCollectionLike(val.getClass())
 				&& targetCollectionType != null && targetComponentType != null) {
+			// Convert collection.
 			return convertCollectionOnRead(val, targetCollectionType, targetComponentType, readConverter);
 		} else if (val.getClass().equals(Blob.class) && targetCollectionType == byte[].class) {
+			// Special case where a single value (Blob) becomes a collection (byte[]).
 			return (T)((Blob) val).toByteArray();
 		}
+		// Convert single value.
 		return (T) readConverter.apply(val, targetComponentType);
 	}
 
 	private <T> T convertCollectionOnRead(Object val, Class targetCollectionType, TypeInformation targetComponentType, BiFunction<Object, TypeInformation<?>, ?> readConverter) {
 		try {
 			List elements;
-
 			if (val.getClass().isArray()) {
 				elements = Collections.singletonList(val);
 			}

--- a/spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/core/convert/DefaultDatastoreEntityConverterTests.java
+++ b/spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/core/convert/DefaultDatastoreEntityConverterTests.java
@@ -103,7 +103,8 @@ public class DefaultDatastoreEntityConverterTests {
 	public void readTest() {
 		byte[] bytes = { 1, 2, 3 };
 		Key otherKey = Key.newBuilder("testproject", "test_kind", "test_name").build();
-		Entity entity = getEntityBuilder()
+		// entity as it would come back from the backend / client library.
+		Entity datastoreEntity = getEntityBuilder()
 				.set("durationField", "PT24H")
 				.set("stringField", "string value")
 				.set("boolField", true)
@@ -115,22 +116,26 @@ public class DefaultDatastoreEntityConverterTests {
 				.set("intField", 99)
 				.set("enumField", "WHITE")
 				.set("keyField", otherKey)
+				.set("byteArrayField", Blob.copyFrom(bytes))
 				.build();
-		TestDatastoreItem item = ENTITY_CONVERTER.read(TestDatastoreItem.class, entity);
+		// Plain Java Object that the user expects to operate on.
+		TestDatastoreItem userItem = ENTITY_CONVERTER.read(TestDatastoreItem.class, datastoreEntity);
 
-		assertThat(item.getDurationField()).as("validate duration field").isEqualTo(Duration.ofDays(1));
-		assertThat(item.getStringField()).as("validate string field").isEqualTo("string value");
-		assertThat(item.getBoolField()).as("validate boolean field").isTrue();
-		assertThat(item.getDoubleField()).as("validate double field").isEqualTo(3.1415D);
-		assertThat(item.getLongField()).as("validate long field").isEqualTo(123L);
-		assertThat(item.getLatLngField()).as("validate latLng field")
+		assertThat(userItem.getDurationField()).as("validate duration field").isEqualTo(Duration.ofDays(1));
+		assertThat(userItem.getStringField()).as("validate string field").isEqualTo("string value");
+		assertThat(userItem.getBoolField()).as("validate boolean field").isTrue();
+		assertThat(userItem.getDoubleField()).as("validate double field").isEqualTo(3.1415D);
+		assertThat(userItem.getLongField()).as("validate long field").isEqualTo(123L);
+		assertThat(userItem.getLatLngField()).as("validate latLng field")
 				.isEqualTo(LatLng.of(10, 20));
-		assertThat(item.getTimestampField()).as("validate timestamp field")
+		assertThat(userItem.getTimestampField()).as("validate timestamp field")
 				.isEqualTo(Timestamp.ofTimeSecondsAndNanos(30, 40));
-		assertThat(item.getBlobField()).as("validate blob field").isEqualTo(Blob.copyFrom(bytes));
-		assertThat(item.getIntField()).as("validate int field").isEqualTo(99);
-		assertThat(item.getEnumField()).as("validate enum field").isEqualTo(TestDatastoreItem.Color.WHITE);
-		assertThat(item.getKeyField()).as("validate key field").isEqualTo(otherKey);
+		assertThat(userItem.getBlobField()).as("validate blob field").isEqualTo(Blob.copyFrom(bytes));
+		assertThat(userItem.getIntField()).as("validate int field").isEqualTo(99);
+		assertThat(userItem.getEnumField()).as("validate enum field").isEqualTo(TestDatastoreItem.Color.WHITE);
+		assertThat(userItem.getKeyField()).as("validate key field").isEqualTo(otherKey);
+
+		assertThat(userItem.getByteArrayField()).as("validate byte array field").isEqualTo(bytes);
 	}
 
 	@Test

--- a/spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/core/convert/DefaultDatastoreEntityConverterTests.java
+++ b/spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/core/convert/DefaultDatastoreEntityConverterTests.java
@@ -116,7 +116,6 @@ public class DefaultDatastoreEntityConverterTests {
 				.set("intField", 99)
 				.set("enumField", "WHITE")
 				.set("keyField", otherKey)
-				.set("byteArrayField", Blob.copyFrom(bytes))
 				.build();
 		// Plain Java Object that the user expects to operate on.
 		TestDatastoreItem userItem = ENTITY_CONVERTER.read(TestDatastoreItem.class, datastoreEntity);
@@ -134,6 +133,18 @@ public class DefaultDatastoreEntityConverterTests {
 		assertThat(userItem.getIntField()).as("validate int field").isEqualTo(99);
 		assertThat(userItem.getEnumField()).as("validate enum field").isEqualTo(TestDatastoreItem.Color.WHITE);
 		assertThat(userItem.getKeyField()).as("validate key field").isEqualTo(otherKey);
+	}
+
+	@Test
+	public void readTestByteArray() {
+		byte[] bytes = { 1, 2, 3 };
+
+		// Datastore Entity from the backend / client library.
+		Entity datastoreEntity = getEntityBuilder()
+				.set("byteArrayField", Blob.copyFrom(bytes))
+				.build();
+		// Plain Java Object that the user expects to operate on.
+		TestDatastoreItem userItem = ENTITY_CONVERTER.read(TestDatastoreItem.class, datastoreEntity);
 
 		assertThat(userItem.getByteArrayField()).as("validate byte array field").isEqualTo(bytes);
 	}

--- a/spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/core/convert/DefaultDatastoreEntityConverterTests.java
+++ b/spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/core/convert/DefaultDatastoreEntityConverterTests.java
@@ -103,7 +103,7 @@ public class DefaultDatastoreEntityConverterTests {
 	public void readTest() {
 		byte[] bytes = { 1, 2, 3 };
 		Key otherKey = Key.newBuilder("testproject", "test_kind", "test_name").build();
-		// entity as it would come back from the backend / client library.
+		// Datastore Entity from the backend / client library.
 		Entity datastoreEntity = getEntityBuilder()
 				.set("durationField", "PT24H")
 				.set("stringField", "string value")

--- a/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/DatastoreRepositoryExample.java
+++ b/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/DatastoreRepositoryExample.java
@@ -16,6 +16,7 @@
 
 package com.example;
 
+import java.nio.charset.StandardCharsets;
 import java.time.LocalDate;
 import java.time.Month;
 import java.util.Arrays;
@@ -69,6 +70,7 @@ public class DatastoreRepositoryExample {
 							new Album("b", LocalDate.of(2018, Month.FEBRUARY, 12)))));
 			Singer richardRoe = new Singer("singer3", "Richard", "Roe",
 					new HashSet<>(Arrays.asList(new Album("c", LocalDate.of(2000, Month.AUGUST, 31)))));
+			richardRoe.setPassword("insecure".getBytes(StandardCharsets.UTF_8));
 
 			this.singerRepository.saveAll(Arrays.asList(janeDoe, richardRoe));
 

--- a/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/DatastoreRepositoryExample.java
+++ b/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/DatastoreRepositoryExample.java
@@ -70,7 +70,7 @@ public class DatastoreRepositoryExample {
 							new Album("b", LocalDate.of(2018, Month.FEBRUARY, 12)))));
 			Singer richardRoe = new Singer("singer3", "Richard", "Roe",
 					new HashSet<>(Arrays.asList(new Album("c", LocalDate.of(2000, Month.AUGUST, 31)))));
-			richardRoe.setPassword("insecure".getBytes(StandardCharsets.UTF_8));
+			richardRoe.setMessage("Hello, dear fans!".getBytes(StandardCharsets.UTF_8));
 
 			this.singerRepository.saveAll(Arrays.asList(janeDoe, richardRoe));
 

--- a/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/Singer.java
+++ b/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/Singer.java
@@ -64,6 +64,8 @@ public class Singer {
 
 	private Set<Album> albums;
 
+	private byte[] password;
+
 	public Singer() {
 	}
 
@@ -130,6 +132,14 @@ public class Singer {
 		this.albums = albums;
 	}
 
+	public byte[] getPassword() {
+		return password;
+	}
+
+	public void setPassword(byte[] password) {
+		this.password = password;
+	}
+
 	@Override
 	public boolean equals(Object o) {
 		if (this == o) {
@@ -162,6 +172,7 @@ public class Singer {
 				+ ((this.personalInstruments == null) ? ""
 						: Strings.join(this.personalInstruments.stream()
 								.map(x -> x.getType()).collect(Collectors.toList()), ','))
+				+ ((this.password == null) ? "" : ", Password (don't actually ever do this): " + new String(password) )
 				+ '}';
 	}
 

--- a/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/Singer.java
+++ b/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/Singer.java
@@ -172,7 +172,7 @@ public class Singer {
 				+ ((this.personalInstruments == null) ? ""
 						: Strings.join(this.personalInstruments.stream()
 								.map(x -> x.getType()).collect(Collectors.toList()), ','))
-				+ ((this.message == null) ? "" : ", Message: " + new String(message) )
+				+ ((this.message == null) ? "" : ", Message: " + new String(message))
 				+ '}';
 	}
 

--- a/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/Singer.java
+++ b/spring-cloud-gcp-samples/spring-cloud-gcp-data-datastore-sample/src/main/java/com/example/Singer.java
@@ -64,7 +64,7 @@ public class Singer {
 
 	private Set<Album> albums;
 
-	private byte[] password;
+	private byte[] message;
 
 	public Singer() {
 	}
@@ -132,12 +132,12 @@ public class Singer {
 		this.albums = albums;
 	}
 
-	public byte[] getPassword() {
-		return password;
+	public byte[] getMessage() {
+		return message;
 	}
 
-	public void setPassword(byte[] password) {
-		this.password = password;
+	public void setMessage(byte[] message) {
+		this.message = message;
 	}
 
 	@Override
@@ -172,7 +172,7 @@ public class Singer {
 				+ ((this.personalInstruments == null) ? ""
 						: Strings.join(this.personalInstruments.stream()
 								.map(x -> x.getType()).collect(Collectors.toList()), ','))
-				+ ((this.password == null) ? "" : ", Password (don't actually ever do this): " + new String(password) )
+				+ ((this.message == null) ? "" : ", Message: " + new String(message) )
 				+ '}';
 	}
 

